### PR TITLE
Adding alt text to images

### PR DESCRIPTION
**Pull request by SalazarJosh**
# Overview
This PR adds alt text to images. Content editors are adding alt text to images but we are not always pulling them into the images. We don't want to pull alt text into ALL images, though, since some are decorative.

Several GraphQL queries needed to be updated to pull in the alt text. Once complete, the components were updated to pull that alt text and apply it.

If there is no alt text provided by the CMS, the image is marked as decorative with an empty alt.

The following JIRA ticket is an amalgamation of 11 sub-tasks that cover alt text on various templates / components.

> This pull request resolves [WEBSITE-17](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-17).

## Anything else?
I also took the time to update the `PageHeader` component ([example](http://localhost:8000/locations-and-hours)). It was previously using two elements to create the image, one for desktop and one for mobile. The desktop element was using a `background-image` css property so we couldn't easily put alt text on that. It's now all one element and has specific styling for desktop and mobile view. This update should not change how the image looks on the page.

Most of these components were getting image information from the GraphQL tree `relationships.field_media_image.relationships.field_media_image`. The alt text lives in `relationships.field_media_image.field_media_image.alt` so I ambiguated the tree up to `relationships.field_media_image` for the image data and then further refined it from there.

## Testing
**Check the following pages to make sure they have alt text on the images:**
[WEBSITE-43](https://mlit.atlassian.net/browse/WEBSITE-43)
- [x] [Location visit template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/hatcher-library)
- [x] [Building visit template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/shapiro-library)
- [x] [Room visit template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/computer-and-video-game-archive)

[WEBSITE-44](https://mlit.atlassian.net/browse/WEBSITE-44)
- [x] [Room body-width](https://deploy-preview-308--future-wwwlib-previews.netlify.app/visit-and-study/creation-and-learning-spaces/conservation-lab)
- [x] [Location body-width](https://deploy-preview-308--future-wwwlib-previews.netlify.app/visit-and-study/creation-and-learning-spaces/shapiro-design-lab/pie-space)
- [x] [Basic body-width](https://deploy-preview-308--future-wwwlib-previews.netlify.app/visit-and-study/computing-and-technology/specialized-technology/anatomage-table)
- [x] [Room full-width](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/hatcher-library/hatcher-north-information-services-desk)
- [x] [Location full-width](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/hatcher-library/ask-librarian-desk)

[WEBSITE-45](https://mlit.atlassian.net/browse/WEBSITE-45)
- [x] [News page template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/about-us/news/librarian-awards-2024)

[WEBSITE-46](https://mlit.atlassian.net/browse/WEBSITE-46)
- [x] [Events page template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/visit-and-study/events-and-exhibits/today-and-upcoming/being-mixed-race-mono-racially-organized)

[WEBSITE-47](https://mlit.atlassian.net/browse/WEBSITE-47)
- [x] [Horizontal destination panel template](https://deploy-preview-308--future-wwwlib-previews.netlify.app/visit-and-study/cafes-and-wellbeing/lactation-rooms)

**The following pages should NOT have alt text on the panels**
[WEBSITE-48](https://mlit.atlassian.net/browse/WEBSITE-48)
- [x] [Standard card panel](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/clark-library/collections)

[WEBSITE-49](https://mlit.atlassian.net/browse/WEBSITE-49)
- [x] [Address and Hours card panel](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours)

- Make sure the PR is consistent in these browsers:
  - [x] Chrome
  - [x] Firefox
  - [ ] Safari  (the assignee was not able to test the pull request in this browser)
  - [x] Edge
- Run accessibility tests:
  - [x] ARC Toolkit
  - [x] axe DevTools

### CODE REVIEW

File: src/reusable/card-image.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import { SPACING } from '../reusable';
 
-export default function CardImage ({ image }) {
+export default function CardImage ({ image, alt }) {
   return (
     <GatsbyImage
       image={image}
-      alt={image.alt || ''}
+      alt={alt || ''}
```
Comment (erinesullivan) [inline src/reusable/card-image.js:10]:
```suggestion
export default function CardImage ({ image, alt = '' }) {
  return (
    <GatsbyImage
      image={image}
      alt={alt}
```

Not a blocker, the `alt` parameter could be defined with the default value of an empty string.

File: src/components/page-header.js [hunk 2]
```diff
@@ -57,28 +58,17 @@ export default function PageHeader ({ breadcrumb, title, summary, image, ...rest
...
-              />
               <GatsbyImage
                 image={imageData}
                 css={{
+                  flex: '0 1 50%',
                   margin: `0 -${SPACING.M}`,
                   [MEDIA_QUERIES.LARGESCREEN]: {
-                    display: 'none'
+                    margin: 0,
+                    maxHeight: '350px'
```
Comment (erinesullivan) [inline src/components/page-header.js:68]:
```suggestion
```
The `maxHeight` will not stretch the image to 100% height when viewing large screen, and the text begins to wrap (smaller than ~1285px). See Computer and Video Game Archive:
* [Preview](https://deploy-preview-308--future-wwwlib-previews.netlify.app/locations-and-hours/computer-and-video-game-archive)
* [Live](https://lib.umich.edu/locations-and-hours/computer-and-video-game-archive)

**Review by erinesullivan — CHANGES_REQUESTED** [review]
Just one change to `src/components/page-header.js`, unless this was intentional? The other comments are just nit-picky code improvements that are non-blocking. Everything else looks fine.

File: src/reusable/card-image.js (hunk 1 above)
Comment (SalazarJosh) [reply src/reusable/card-image.js]: Added and changed. Committing the suggestion from the github UI isn't erroring out so I changed it locally and pushed.

File: src/templates/landing.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ export default function LandingTemplate ({ data }) {
         summary={body ? body.summary : null}
         image={
           relationships.field_media_image
-          && relationships.field_media_image.relationships.field_media_image
+          && relationships.field_media_image
         }
```
[the remark this answers is not in the document]
Comment (SalazarJosh) [reply src/templates/landing.js:33]: MUCH cleaner! Lib's GraphQL has a lot of redundant nesting and is difficult to navigate and understand, I'm finding.

File: src/components/page-header.js (hunk 2 above)
Comment (SalazarJosh) [reply src/components/page-header.js:68]: Thanks, Erin. This was intentional to have close-to-parity with how this image is currently handled. It's currently a background-image and I'm changing it to an  so we can add alt text. The downside, as you've pointed out, is how it looks on the Computer and Video Game Archive page. I'll rework this so it's as close to prod as possible and report back soon...

**Review by erinesullivan — APPROVED** [review]
👍🏻 Everything looks good!

**Pull request merged**